pkgs/development/interpreters/ruby/default.nix

```
@@ -113,6 +110,14 @@ let
             })
           ];
 
+        cargoRoot = if yjitSupport then "yjit" else null;
+
+        cargoDeps = if yjitSupport then rustPlatform.fetchCargoTarball {
+          inherit src;
+          sourceRoot = "${pname}-${version}/${cargoRoot}";
+          sha256 = cargoSha256;
+        } else null;
```

```suggestion
        cargoRoot = lib.optionalString yjitSupport "yjit";

        cargoDeps = if yjitSupport then rustPlatform.fetchCargoTarball {
          inherit src;
          sourceRoot = "${pname}-${version}/${cargoRoot}";
          sha256 = cargoSha256;
        } else null;
```
since we are doing a mass rebuild anyway we can also use our lib help functions. Not sure what is the right one for cargoDeps.